wireapp/wire-android
# feat(migration): read and store server config [AR-2384]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2384" title="AR-2384" target="_blank"><img alt="Subtask" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10816?size=medium" />AR-2384</a>  read the scala app server config
  </td></table>
  <br />
 








----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

To make a successful migration, the first step is to retrieve the server config, fetch all missing data and store it in a new database.

### Solutions

Created new use case to execute server config migration. 
Added global DataStore where now the info about the migration completion is stored.
Implemented ScalaBackendPreferences and DAO to retrieve the server config data from the old app.
Migration process organised in the MigrationManager, which also can check whether the migration is required.
MigrationWorker returns the Flow to observe migration result so the app can navigate after successful one or handle errors if necessary in the future.

### Dependencies (Optional)

Needs releases with:

- https://github.com/wireapp/kalium/pull/1017

### Testing

#### Test Coverage (Optional)


#### How to Test

Install AR as an update to the old scala app, then at the launch the migration should happen.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
